p2p/dhtdiscovery/discovery.go

```diff
@@ -173,13 +194,27 @@ func New(h DiscoveryHost, opts ...Opt) (*Discovery, error) {
 		advertiseInterval:   time.Minute,
 		advertiseRetryDelay: time.Minute,
 		findPeersRetryDelay: time.Minute,
+		minBackoff:          60 * time.Second,
+		maxBackoff:          time.Hour,
+		minConnBackoff:      10 * time.Second,
+		maxConnBackoff:      time.Hour,
+		dialTimeout:         2 * time.Minute,
 	}
 	for _, opt := range opts {
 		opt(&d)
 	}
 	if len(d.bootnodes) == 0 {
 		d.logger.Warn("no bootnodes in the config")
 	}
+	rngSrc := rand.NewSource(rand.Int63())
+	cacheSize := 100
+	var err error
+	d.connBackoff, err = backoff.NewBackoffConnector(h, cacheSize, d.dialTimeout,
+		backoff.NewExponentialBackoff(d.minConnBackoff, d.maxConnBackoff, backoff.FullJitter,
+			time.Second, 5.0, 0, rand.New(rngSrc)))
```

Comment:
That seems a bit convoluted, how about just passing a small wrapper around randv2?

```
type rngSource struct{}

func (r rngSource) Int63() int64 {
	return randv2.Int64()
}

func (r rngSource) Seed(seed int64) {}
```

```suggestion
	cacheSize := 100
	var err error
	d.connBackoff, err = backoff.NewBackoffConnector(h, cacheSize, d.dialTimeout,
		backoff.NewExponentialBackoff(d.minConnBackoff, d.maxConnBackoff, backoff.FullJitter,
			time.Second, 5.0, 0, rngSource{}))
```

Reply:
Fixed